prog/vm/nexus.rb

```diff
@@ -437,24 +437,21 @@ def before_run
   end
 
   label def wait_sshable
-    addr = vm.ephemeral_net4
-
-    # Alas, our hosting environment, for now, doesn't support IPv6, so
-    # only check SSH availability when IPv4 is available: a
-    # unistacked IPv6 server will not be checked.
-    #
-    # I considered removing wait_sshable altogether, but (very)
-    # occasionally helps us glean interesting information about boot
-    # problems.
-    hop_create_billing_record unless addr
+    case host.sshable.cmd("common/bin/daemonizer --check wait_sshable_#{q_vm}")
+    when "Succeeded"
+      vm.vm_host.sshable.cmd("common/bin/daemonizer --clean wait_sshable_#{q_vm}")
 
-    begin
-      Socket.tcp(addr.to_s, 22, connect_timeout: 1) {}
-    rescue SystemCallError
-      nap 1
+      hop_create_billing_record
+    when "NotStarted", "Failed"
+      # I considered removing wait_sshable altogether, but (very)
+      # occasionally helps us glean interesting information about boot
```

Comment:
It also helped some issues where IPv6 was connectable, but IPv4 had some setup problems due to the guest OS, etc. I thought we were sacrificing that part, right?

Reply:
I suppose we are, since, unfortunately, there's not a straightforward bank of ipv4 addresses to put the source address from. I suppose we could play around with an address range that is not usually used, such as [240/4](https://datatracker.ietf.org/doc/draft-schoen-intarea-unicast-240/), or possibly, a link-local address.

Reply:
I choose to not solve this for now. Let's see if IPv6 verify_sshable check is enough for now. As long as we add new factors into the matrix with E2E testing + a development cycle, we should be able to catch the issues with IPv4 stack before introducing it to the production, anyway. If we find a scenario that is consistently causing issues, we can think about it.